android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java

```diff
@@ -0,0 +1,73 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.auth.User;
+import org.fossasia.openevent.dbutils.LiveRealmDataObject;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class UserProfileActivityViewModel extends ViewModel {
+    private final CompositeDisposable compositeDisposable;
+    private RealmDataRepository realmRepo;
+    private MutableLiveData<Integer> loadUserFromNetworkResponse;
+    public static final int RETRIEVE_SUCCESSFUL = 1;
+    public static final int RETRIEVE_UNSUCCESSFUL = 2;
+    public static final int RETRIEVE_COMPLETE = 3;
+    private LiveRealmDataObject<User> liveRealmDataObject;
+
+    public UserProfileActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        compositeDisposable = new CompositeDisposable();
+    }
+
+    public void eraseUserData() {
+        realmRepo.clearUserData();
+    }
+
+    public LiveData<User> getUser() {
+        if(liveRealmDataObject == null) {
+            liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getUser());
+            Timber.d("User data loaded from disk");
+        }
+        return  liveRealmDataObject;
+    }
+
+    public LiveData<Integer> loadUserFromNetwork(long id) {
+        if(loadUserFromNetworkResponse == null){
+            loadUserFromNetworkResponse = new MutableLiveData<>();
+        }
+        Disposable disposable = APIClient.getOpenEventAPI()
+                .getUser(id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(user -> {
+                    realmRepo.saveUser(user).subscribe();
+                    loadUserFromNetworkResponse.setValue(RETRIEVE_SUCCESSFUL);
+                    Timber.d("User data saved in database");
+                    },
+                        throwable -> {
```

Comment:
indentation is off here in the onError.

Reply:
that's not off, IDE automatically takes you there when you hit enter, that's the way Android Studio recommend it that means. I believe keep it like that would be good

Reply:
It is off `},` should be either 1 indent less than previous indentation or previous should be 1 more

Secondly, throwable should be on previous line and its block should be one indent less

Reply:
okay i will make the changes 👍